slackBot.js

```diff
@@ -5,155 +5,163 @@ const { WebClient }             = require('@slack/web-api'),
+  async sendForm(messageJson, modalJson, channelId, actionId, callback){
+    try {
+      const messageJsonBlock = {...{channel: channelId}, ...messageJson};
+      const response = await this.webClient.chat.postMessage(messageJsonBlock);
+      
+      // sends the form when the user clicks the open form button
+      this.slackInteractiveMessages.action({ actionId: actionId}, async (payload) => {
+          await this.webClient.views.open({
+              trigger_id: payload.trigger_id,
+              view: modalJson
+          }) 
+          //returns the server response when we send the button (response) as well as the payload generated by the server when they click it.
+          callback(response,payload);
+      });
+    } catch(err) {
+      console.log(err);
+    }
+  }
+
+  // uploads a file to the specified channel, you need to provide the channelId and filepath
+  async sendFile(channelId, filePath, callback){
+    try {
+      const response = await this.webClient.files.upload({channels: channelId, file: await createReadStream(filePath)});
+      callback(response);
+    } catch(err){
+      console.log(err);
+    }
+  }
+
+  sendMessageToChannel(channelId, message) {
+    const data = {
+      channel: channelId,
+      text: message
+    };
+    return this.webClient.chat.postMessage(data);
+  }
+
+  async sendMessageToAllChannels(message) {
+    const res = await this.webClient.conversations.list();
+    const sendMessages = res.channels
+      .filter(channel => channel.is_member)
+      .map(channel => this.sendMessageToChannel(channel.id, message))
+    return Promise.all(sendMessages);
+  }
+
+  async sendMessageChannel(channelId, message) {
+    return this.sendMessageToChannel(channelId, message);
+  }
+
+  async getChannels() {
+    const ch = await this.webClient.conversations.list();
+    return ch.channels
+      .filter(channel => channel.is_member)
+      .map(channel => ({
+        id: channel.id,
+        name: channel.name,
+      })); 
+  }
+
+  getMembers() {
+      return this.webClient.users.list();
+  }
+
+
+  stop(slackEventsPort = 3000, slackInteractiveMessagesPort = 3001){
+    this.slackEvents.stop(slackEventsPort);
+    this.slackInteractiveMessages.stop(slackInteractiveMessagesPort);
+  }
+
+  // can add a custom command
+  addCustomCommand(command, callback){
+    this.slackEvents.on('message', (message) => {
+      if (message.text.split(' ')[0] == command) {
+        callback(message);
       }
-    }
+    });
+  }
 
-    // uploads a file to the specified channel, you need to provide the channelId and filepath
-    async sendFile(channelId, filePath, callback){
-      try {
-        const response = await this.webClient.files.upload({channels: channelId, file: await createReadStream(filePath)});
-        callback(response);
-      } catch(err){
-        console.log(err);
+  listenForFiles(callback) {
+    this.slackEvents.on('message', (message) => {
+      if('files' in message){
```

Comment:
if I read it correctly, it would trigger the callback if the user's input message string include "files". I'm assuming slack event have a different listener for sending files except checking the files string in the "message" listener, that we need to figure out.